Utilities/StaticAnalyzers/src/UnnecessaryMutableChecker.cpp

```diff
@@ -46,6 +46,12 @@ namespace clangcms {
       if (!Field->isMutable()) {
         continue;
       }
+
+      // == Skip non-public mutables, we deal with them elsewhere
```

Comment:
Does "elsewhere" refer to `PublicMutableChecker`, or something else?

Reply:
To `MutableMemberModificationChecker`. This comment is indeed a bit misleading, since these two checkers look for similar, but different, things.

Reply:
I'm confused. Why non-public mutables should not be checked here?

Reply:
@makortel , non-private (proteced and public) mutable member are flaged by `PublicMutableChecker`. As non-private mutables can be modified on other classes (e.g. protected in their drived classes and public in any other class) so it is bit hard to find is a mutable is useless (not modified in any const method). That is why `UnnecessaryMutableChecker` only checks for private mutables. non-private multbles are anyway flaged as thread safety issue by `PublicMutableChecker`

Reply:
Right, that's along what I thought too. But the comment and code here skip non-public mutables, i.e. it checks only public mutables. Should the code here be then
```cpp
// == Skip non-private mutables, we deal with them elsewhere
if (Field->getAccess() != clang::AS_private) {
 return;
}
```
?

(being more specific with "elsewhere" would helpful though)

Reply:
Yes. Thanks for spotting! I have updated the comment as well.